packages/combobox/src/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,740 @@
+          event.preventDefault();
+        }
+      }}
+      tgphRef={context.triggerRef}
+    >
+      <RefToTgphRef>
+        <TelegraphButton.Root
+          id={context.triggerId}
+          variant="outline"
+          w="full"
+          h="full"
+          color={context.errored ? "red" : "gray"}
+          justify="space-between"
+          // Accessibility attributes
+          role="combobox"
+          aria-label={getLabelString()}
+          aria-controls={context.contentId}
+          aria-expanded={context.open}
+          aria-haspopup="listbox"
+          // Custom attributes
+          data-tgph-combobox-trigger
+          data-tgph-comobox-trigger-open={context.open}
+        >
+          <TriggerValue size={size} />
+          <TelegraphButton.Icon
+            as={motion.div}
+            icon={Lucide.ChevronDown}
+            animate={{ rotate: context.open ? "180deg" : "0deg" }}
+            transition={{ duration: 0.2, type: "spring", bounce: 0 }}
+            aria-hidden
+          />
+        </TelegraphButton.Root>
+      </RefToTgphRef>
+    </TelegraphMenu.Anchor>
+  );
+};
+
+type ContentProps<T extends TgphElement> = TgphComponentProps<
+  typeof TelegraphMenu.Content<T>
+>;
+
+const Content = <T extends TgphElement>({
+  tgphRef,
+  style,
+  children,
+  ...props
+}: ContentProps<T>) => {
+  const context = React.useContext(ComboboxContext);
+  const hasInteractedOutside = React.useRef(false);
+  const composedRef = useComposedRefs<unknown>(tgphRef, context.contentRef);
+
+  const internalContentRef = React.useRef(null);
+
+  const [height, setHeight] = React.useState(0);
+  const [initialAnimationComplete, setInitialAnimationComplete] =
+    React.useState(false);
+
+  const setHeightFromContent = React.useCallback(
+    (element: Element) => {
+      // Set the initial height of the content after the animation completes
+      const rect = element.getBoundingClientRect();
+      if (rect) {
+        setHeight(rect.height);
+      }
+
+      if (!initialAnimationComplete) {
+        setInitialAnimationComplete(true);
+      }
+    },
+    [initialAnimationComplete],
+  );
+
+  React.useEffect(() => {
+    const observer = new ResizeObserver((entries) => {
+      for (const entry of entries) {
+        const element = entry.target;
+        setHeightFromContent(element);
+      }
+    });
+    // Attatch the observer once the initial animation completes
+    // and the content ref is available
+    if (internalContentRef.current && initialAnimationComplete) {
+      observer.observe(internalContentRef.current);
+    }
+
+    return () => observer.disconnect();
+  }, [initialAnimationComplete, setHeightFromContent]);
+
+  // Reset the animation complete state when the combobox is closed
+  React.useEffect(() => {
+    if (initialAnimationComplete === true && context.open === false) {
+      setInitialAnimationComplete(false);
+    }
+  }, [context.open, initialAnimationComplete]);
+
```

Comment:
Logic for creating the smooth height animation change as you're searching inside of the combobox.